metabase/metabase
# Make notebook native preview sidebar resizable

This task belongs to the milestone 3 of #40254.

### Description
This PR introduces a resizable native query preview sidebar using a `ResizableBox` component from `react-resize` library.

Te biggest challenge while working on this task was to conditionally show either the resizable sidebar or the regular, full-width native query preview - all based on the current window width. The only solution I came up with was to literally display two distinct wrappers around the `NativeQueryPreviewSidebar` component. One is the resizable box and the other one is HTML 5 `aside`.

This required moving some of the logic from the CSS module directly to this component.

This PR also introduces a new key (`nativePreviewSidebarWidth`) to the redux store's `qb.uiControls` object.

### How to verify
1. Open a notebook view
2. Show SQL sidebar
3. It should initially be set to 428px
4. Try making it smaller - it should not be possible
5. Try resizing it to a bigger size
6. Close the sql preview
7. Open it again
8. Sidebar should've retained the last set width instead of the initial 428px
9. The maximum width of the sidebar is dictated by the min width of the notebook editor (which is 640px)

### Demo
https://github.com/metabase/metabase/assets/31325167/50053cba-3794-4f37-a598-b81248a2e494

--- 
> [!Important]
> I will cover this behavior with E2E tests in the upcoming PR that will persist both the sidebar state and the sidebar width to the backend user setting.

Please note that we've been able to re-enable the crucial E2E test for the CI thanks to changes in this PR. ReplayIO does not support nested media queries in the Chrome version yet. We got rid of the CSS module altogether so there's nothing stoping us from running this test in the CI again.
https://github.com/metabase/metabase/pull/40984/commits/e76e8de857b43654624598cd60ce4f6198a4687f
